openc3/lib/openc3/script/script_runner.rb

```diff
@@ -0,0 +1,238 @@
+# encoding: ascii-8bit
+
+# Copyright 2022 OpenC3, Inc.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+module OpenC3
+  module Script
+    private
```

Comment:
Is this just a white list of allowed methods? Why is this necessary when it worked before (except without authentication)?

Reply:
This is new Ruby client code to spawn scripts and interact with them.  Before this code was only in Javascript.